### PR TITLE
cz-pulsar: Alphafold rules

### PR DESCRIPTION
**Pull request by martindemko**
Hi guys,
I'm trying to handle different alphafold database versions for CESNET clusters. Not exactly sure if this is the correct way to do it, still new with TPV. If I look to you as totally lost, please, advise me a little bit.
Thanks,
Martin

### CODE REVIEW

File: files/galaxy/tpv/destinations.yml.j2 [hunk 1]
```diff
@@ -283,6 +283,16 @@ destinations:
     scheduling:
       require:
         - cz-pulsar
+    rules:
+      - if: "alphafold2/alphafold" in tool.id and helpers.tool_version_lt(tool, '2.3') and helpers.tool_version_gte(tool, '2.1.2+galaxy0')
+        env: 
+          ALPHAFOLD_DB: "/storage/brno11-elixir/projects/alphafold/alphafold.db-2.2"
+        params:
+          singularity_volumes: "$job_directory:ro,$tool_directory:ro,$job_directory/outputs:rw,$working_directory:rw,/cvmfs/data.galaxyproject.org:ro,/storage/praha5-elixir/home/galaxyeu:/home/galaxyeu,$SCRATCHDIR,$ALPHAFOLD_DB:/data/2.2:ro"
```
Comment (kysrpex) [inline files/galaxy/tpv/destinations.yml.j2:291]:
```suggestion
      - if: tool.id.startswith("toolshed.g2.bx.psu.edu/repos/galaxy-australia/alphafold2/alphafold") and helpers.tool_version_gte(tool, "2.1.2+galaxy0") and helpers.tool_version_lt(tool, "2.3")
        env:
          ALPHAFOLD_DB: "/storage/brno11-elixir/projects/alphafold/alphafold.db-2.2"
        params:
          singularity_volumes: "$job_directory:ro,$tool_directory:ro,$job_directory/outputs:rw,$working_directory:rw,/cvmfs/data.galaxyproject.org:ro,$SCRATCHDIR,/storage/praha5-elixir/home/galaxyeu:/home/galaxyeu,$ALPHAFOLD_DB:/data/2.2:ro"
```

File: files/galaxy/tpv/destinations.yml.j2 [hunk 2]
```diff
@@ -283,6 +283,16 @@ destinations:
...
+      - if: "alphafold2/alphafold" in tool.id and helpers.tool_version_gt(tool, '2.3.1+galaxy0')
+        params:
+          singularity_volumes: "$job_directory:ro,$tool_directory:ro,$job_directory/outputs:rw,$working_directory:rw,/cvmfs/data.galaxyproject.org:ro,/storage/praha5-elixir/home/galaxyeu:/home/galaxyeu,$SCRATCHDIR,$ALPHAFOLD_DB:/data/2.3:ro"
+
```
Comment (kysrpex) [inline files/galaxy/tpv/destinations.yml.j2:295]:
```suggestion
      - if: tool.id.startswith("toolshed.g2.bx.psu.edu/repos/galaxy-australia/alphafold2/alphafold") and helpers.tool_version_gte(tool, "2.3")
        params:
          singularity_volumes: "$job_directory:ro,$tool_directory:ro,$job_directory/outputs:rw,$working_directory:rw,/cvmfs/data.galaxyproject.org:ro,$SCRATCHDIR,/storage/praha5-elixir/home/galaxyeu:/home/galaxyeu,$ALPHAFOLD_DB:/data/2.3:ro"

```

**Review by kysrpex — CHANGES_REQUESTED** [review]

**Comment by kysrpex** [comment]
Is Singularity actually reading your environment variables?

I am asking because depending on your setup you may need [an approach similar to ours](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/8d3532faae4719733d626b89e441acccb63acedd/files/galaxy/tpv/tools.yml#L259).

**Comment by martindemko** [comment]
Thanks for the hint, I didn't think about that. However, ALPHAFOLD_DB is used outside of singularity in our case only ALPHAFOLD_USE_GPU is used inside, so I used your solution for it.

**Review by kysrpex — APPROVED** [review]

**Pull request merged**